test/functional/lua/vim_spec.lua

```diff
@@ -794,6 +794,16 @@ describe('lua stdlib', function()
           pcall_err(exec_lua, "vim.fn.nvim_get_current_line()"))
   end)
 
+  it('vim.fn can be called in fast events (if function is allowed)', function()
+    exec_lua([[
+      vim.loop.fs_stat('.', function()
+        vim.g.fnres = vim.fn.iconv('hello', 'utf-8', 'utf-8')
+      end)
+    ]])
+    helpers.sleep(10)
```

Comment:
I think `poke_eventloop` might work here, and is semantically helpful?

Reply:
No, actually. `poke_eventloop` iterates to the next main event, whereas here we want to iterate to the next fast event.

I can modify `poke_eventloop` so it works for both.

Reply:
Switched to a 0-time timer and a single call to `poke_eventloop` appears to work.